### PR TITLE
fix `dirpath` in `log_dir` for CSVLogger

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

replace the rather more robust way of obtaining `log_dir`
see failing legacy testing: https://github.com/Lightning-AI/lightning/actions/runs/3938658456/jobs/6737646117

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @awaelchli @borda @Blaizzy @tchaton

### CODE REVIEW

**Review by lantiga — APPROVED** [review]
Great catch @Borda!

File: src/pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1807,10 +1807,8 @@ def model(self, model: torch.nn.Module) -> None:
     @property
     def log_dir(self) -> Optional[str]:
         if len(self.loggers) > 0:
-            if not isinstance(self.loggers[0], TensorBoardLogger):
-                dirpath = self.loggers[0].save_dir
-            else:
-                dirpath = self.loggers[0].log_dir
+            logger_ = self.loggers[0]
+            dirpath = getattr(logger_, "log_dir", getattr(logger_, "save_dir", None))
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/trainer.py:1810]: Why has log_dir precedence over save_dir? It should be the other way around to be consistent with the way it was before.

**Review by carmocca — APPROVED** [review]
LGTM. I'd like to get @awaelchli's stamp too though

File: src/pytorch_lightning/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -369,7 +369,8 @@ def _save_checkpoint(self, trainer: "pl.Trainer", filepath: str) -> None:
         # notify loggers
         if trainer.is_global_zero:
             for logger in trainer.loggers:
-                logger.after_save_checkpoint(proxy(self))
+                if hasattr(logger, "after_save_checkpoint"):
```
Comment (carmocca) [inline src/pytorch_lightning/callbacks/model_checkpoint.py:372]: @Borda where did you see this fail? The base logger interface includes this method.

Comment (Borda) [reply src/pytorch_lightning/callbacks/model_checkpoint.py:372]: it is the initial log, or you can just run `tests/legacy/simple_classif_training.py` with just install PL without loggers :chicken:

**Review by awaelchli — APPROVED** [review]
I don't understand why you get an attribute error. The attribute/property exists on the base logger and returns None by default. So this shouldn't happen at all. 

Note: Underlying inconsistency will get fixed with https://github.com/Lightning-AI/lightning/issues/14188, but I haven't had the time to complete the work.

File: src/pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (Borda) [reply src/pytorch_lightning/trainer/trainer.py]: :+1:

**Comment by Borda** [comment]
> Underlying inconsistency will get fixed with #14188, but I haven't had the time to complete the work.

this would be great! it comes from creating legacy checkpoints which have not been part of normal CI so adding it here too :otter:

File: src/pytorch_lightning/callbacks/model_checkpoint.py (hunk 2 above)
Comment (carmocca) [reply src/pytorch_lightning/callbacks/model_checkpoint.py:372]: `PYTHONPATH=..:$PYTHONPATH python simple_classif_training.py` also doesn't fail with this change remove

Comment (Borda) [reply src/pytorch_lightning/callbacks/model_checkpoint.py:372]: > The job in the description shows a different error.

which one? 
`AttributeError: 'CSVLogger' object has no attribute 'save_dir'`
and the I continued...

Comment (Borda) [reply src/pytorch_lightning/callbacks/model_checkpoint.py:372]: but if this is fine for you I l remove it :)

**Pull request merged**